postgresql/resource_postgresql_role.go

```diff
@@ -588,14 +595,22 @@ func readAssumeRole(roleConfig pq.ByteaArray) string {
 	return res
 }
 
-func readRoleParameters(roleConfig pq.ByteaArray) *schema.Set {
+func readRoleParameters(roleConfig pq.ByteaArray, existingParams *schema.Set) *schema.Set {
 	params := make([]interface{}, 0)
 	for _, v := range roleConfig {
 		tokens := strings.Split(string(v), "=")
 		if !sliceContainsStr(ignoredRoleConfigurationParameters, tokens[0]) {
+			quote := true
+			for _, p := range existingParams.List() {
+				existingParam := p.(map[string]interface{})
+				if existingParam[roleParameterNameAttr].(string) == tokens[0] {
+					quote = existingParam[roleParameterQuoteAttr].(bool)
+				}
+			}
```

Comment:
Without this block, I was getting bad diffs showing a `false` value being read from Postgres after an apply. The workaround (it feels more like that than a "fix") was to check the configuration for `quote` setting for each parameter and just copy it, defaulting to `true` if it's not found. The only scenario I could imagine this being strange in is an import.

Reply:
Hmm, yeah awkward that setting `Default` on the schema wouldn't be enough...maybe just a quirk of being on an earlier version of the SDK 🤷🏾